distributed/comm/core.py

```diff
@@ -263,7 +263,7 @@ def _raise(error):
             timeout,
             error,
         )
-        raise IOError(msg)
+        raise IOError(msg) from error
```

Comment:
This is one change I have made speculatively and am conscious may not be preferred - e.g. I'm not confident I know what happens with the below snippet (I'm not sure what the value of `error` is there.

https://github.com/dask/distributed/blob/76de87bb49faf00a84947bf0b2c6708c509d89c0/distributed/comm/core.py#L312-L313